probe/awsecs/reporter.go

```diff
@@ -0,0 +1,121 @@
+package awsecs
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/weaveworks/scope/probe/docker"
+	"github.com/weaveworks/scope/report"
+)
+
+// TaskFamily is the key that stores the task family of an ECS Task
+const (
+	TaskFamily = "ecs_task_family"
+)
+
+type taskInfo struct {
+	containerIDs []string
+	family       string
+}
+
+// return map from cluster to map of task arns to task infos
+func getLabelInfo(rpt report.Report) map[string]map[string]*taskInfo {
+	results := make(map[string]map[string]*taskInfo)
+	log.Debug("scanning for ECS containers")
+	for nodeID, node := range rpt.Container.Nodes {
+
+		taskArn, taskArnOk := node.Latest.Lookup(docker.LabelPrefix + "com.amazonaws.ecs.task-arn")
+		cluster, clusterOk := node.Latest.Lookup(docker.LabelPrefix + "com.amazonaws.ecs.cluster")
+		family, familyOk := node.Latest.Lookup(docker.LabelPrefix + "com.amazonaws.ecs.task-definition-family")
+
+		if taskArnOk && clusterOk && familyOk {
```

Comment:
to reduce the nesting in this function, I'd probably push `if !ok { continue }` up after each lookup.